kibiutils not working on 5.5.2

```
> kibi@5.5.2 start /home/sergio/kibi/kibi-internal-merge
> sh ./bin/kibi --dev

/home/sergio/kibi/kibi-internal-merge/node_modules/kibiutils/lib/migrations/migration_runner.js:114
      var _ref = _asyncToGenerator(regeneratorRuntime.mark(function _callee() {
                                   ^

ReferenceError: regeneratorRuntime is not defined
    at /home/sergio/kibi/kibi-internal-merge/node_modules/kibiutils/lib/migrations/migration_runner.js:114:36
    at /home/sergio/kibi/kibi-internal-merge/node_modules/kibiutils/lib/migrations/migration_runner.js:205:6
    at Object.<anonymous> (/home/sergio/kibi/kibi-internal-merge/node_modules/kibiutils/lib/migrations/migration_runner.js:352:2)
    at Module._compile (module.js:570:32)
    at Module._extensions..js (module.js:579:10)
    at Object.require.extensions.(anonymous function) [as .js] (/home/sergio/kibi/kibi-internal-merge/node_modules/babel-register/lib/node.js:152:7)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/home/sergio/kibi/kibi-internal-merge/src/cli/kibi/upgrade.js:7:1)
    at Module._compile (module.js:570:32)
    at loader (/home/sergio/kibi/kibi-internal-merge/node_modules/babel-register/lib/node.js:144:5)
    at Object.require.extensions.(anonymous function) [as .js] (/home/sergio/kibi/kibi-internal-merge/node_modules/babel-register/lib/node.js:154:7)
    at Module.load (module.js:487:32)

npm ERR! Linux 4.10.0-32-generic
npm ERR! argv "/home/sergio/.nvm/versions/node/v6.11.1/bin/node" "/home/sergio/.nvm/versions/node/v6.11.1/bin/npm" "start"
npm ERR! node v6.11.1
npm ERR! npm  v3.10.10
npm ERR! code ELIFECYCLE
npm ERR! kibi@5.5.2 start: `sh ./bin/kibi --dev`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the kibi@5.5.2 start script 'sh ./bin/kibi --dev'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the kibi package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     sh ./bin/kibi --dev
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs kibi
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls kibi
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/sergio/kibi/kibi-internal-merge/npm-debug.log

```

--- a/lib/kibiutils.js
+++ b/lib/kibiutils.js
@@ -1,6 +1,7 @@
 'use strict';
 
 (function () {
+  require("babel-polyfill");
 
   function KibiUtils() {}
 
@@ -34,7 +35,7 @@
           return true;
         }
 
-        var i;
+        var i = void 0;
         if (query.rest_params) {
           for (i = 0; i < query.rest_params.length; i++) {
             if (entityRegex.test(query.rest_params[i].value)) {

--- a/lib/migrations/__tests__/migration.js
+++ b/lib/migrations/__tests__/migration.js
@@ -48,7 +48,7 @@ describe('migrations', function () {
         _sinon2.default.spy(client, 'count');
       });
 
-      it('should build the correct query and return the result', _asyncToGenerator(regeneratorRuntime.mark(function _callee() {
+      it('should build the correct query and return the result', _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee() {
         var migration, index, type, query, result;
         return regeneratorRuntime.wrap(function _callee$(_context) {
           while (1) {
@@ -127,7 +127,7 @@ describe('migrations', function () {
         });
       });
 
-      it('should set default options if options have not been defined', _asyncToGenerator(regeneratorRuntime.mark(function _callee2() {
+      it('should set default options if options have not been defined', _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee2() {
         var migration;
         return regeneratorRuntime.wrap(function _callee2$(_context2) {
           while (1) {
@@ -150,7 +150,7 @@ describe('migrations', function () {
         }, _callee2, _this2);
       })));
 
-      it('should set a default size if no size has been specified', _asyncToGenerator(regeneratorRuntime.mark(function _callee3() {
+      it('should set a default size if no size has been specified', _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee3() {
         var migration;
         return regeneratorRuntime.wrap(function _callee3$(_context3) {
           while (1) {
@@ -173,7 +173,7 @@ describe('migrations', function () {
         }, _callee3, _this2);
       })));
 
-      it('should use the specified size', _asyncToGenerator(regeneratorRuntime.mark(function _callee4() {
+      it('should use the specified size', _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee4() {
         var migration;
         return regeneratorRuntime.wrap(function _callee4$(_context4) {
           while (1) {
@@ -196,7 +196,7 @@ describe('migrations', function () {
         }, _callee4, _this2);
       })));
 
-      it('should use the scroll API to fetch hits', _asyncToGenerator(regeneratorRuntime.mark(function _callee5() {
+      it('should use the scroll API to fetch hits', _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee5() {
         var migration, index, type, query, options, results, i;
         return regeneratorRuntime.wrap(function _callee5$(_context5) {
           while (1) {

--- a/lib/migrations/__tests__/migration_runner.js
+++ b/lib/migrations/__tests__/migration_runner.js
@@ -45,7 +45,7 @@ describe('migrations', function () {
         _createClass(_class, [{
           key: 'count',
           value: function () {
-            var _ref = _asyncToGenerator(regeneratorRuntime.mark(function _callee() {
+            var _ref = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee() {
               return regeneratorRuntime.wrap(function _callee$(_context) {
                 while (1) {
                   switch (_context.prev = _context.next) {
@@ -79,7 +79,7 @@ describe('migrations', function () {
         }, {
           key: 'upgrade',
           value: function () {
-            var _ref2 = _asyncToGenerator(regeneratorRuntime.mark(function _callee2() {
+            var _ref2 = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee2() {
               return regeneratorRuntime.wrap(function _callee2$(_context2) {
                 while (1) {
                   switch (_context2.prev = _context2.next) {
@@ -255,7 +255,7 @@ describe('migrations', function () {
 
       var runner = new _migration_runner2.default(server3, logger);
 
-      it('should stop on 5th iteration', _asyncToGenerator(regeneratorRuntime.mark(function _callee3() {
+      it('should stop on 5th iteration', _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee3() {
         var result;
         return regeneratorRuntime.wrap(function _callee3$(_context3) {
           while (1) {
@@ -298,7 +298,7 @@ describe('migrations', function () {
         _sinon2.default.spy(runner, 'getMigrations');
       });
 
-      it('should execute second iteration if there are still migrations after the first one', _asyncToGenerator(regeneratorRuntime.mark(function _callee4() {
+      it('should execute second iteration if there are still migrations after the first one', _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee4() {
         var result;
         return regeneratorRuntime.wrap(function _callee4$(_context4) {
           while (1) {
@@ -341,7 +341,7 @@ describe('migrations', function () {
         _sinon2.default.spy(runner, 'getMigrations');
       });
 
-      it('should execute migrations in the correct order', _asyncToGenerator(regeneratorRuntime.mark(function _callee5() {
+      it('should execute migrations in the correct order', _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee5() {
         var result, migrations, descriptions;
         return regeneratorRuntime.wrap(function _callee5$(_context5) {
           while (1) {
@@ -399,7 +399,7 @@ describe('migrations', function () {
         _sinon2.default.spy(runner, 'getMigrations');
       });
 
-      it('should execute migrations in the correct order', _asyncToGenerator(regeneratorRuntime.mark(function _callee6() {
+      it('should execute migrations in the correct order', _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee6() {
         var result, migrations, descriptions;
         return regeneratorRuntime.wrap(function _callee6$(_context6) {
           while (1) {
@@ -458,7 +458,7 @@ describe('migrations', function () {
           _sinon2.default.spy(plugin3, 'getMigrations');
         });
 
-        it('cache migrations', _asyncToGenerator(regeneratorRuntime.mark(function _callee7() {
+        it('cache migrations', _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee7() {
           var runner;
           return regeneratorRuntime.wrap(function _callee7$(_context7) {
             while (1) {

--- a/lib/migrations/migration.js
+++ b/lib/migrations/migration.js
@@ -14,9 +14,12 @@ function _asyncToGenerator(fn) { return function () { var gen = fn.apply(this, a
 
 function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
 
+require("babel-polyfill");
+
 /**
  * The base class for migrations.
  */
+
 var Migration = function () {
   /**
    * Creates a new Migration.
@@ -45,7 +48,7 @@ var Migration = function () {
      * Returns the number of objects that can be upgraded by this migration.
      */
     value: function () {
-      var _ref = _asyncToGenerator(regeneratorRuntime.mark(function _callee() {
+      var _ref = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee() {
         return regeneratorRuntime.wrap(function _callee$(_context) {
           while (1) {
             switch (_context.prev = _context.next) {
@@ -74,7 +77,7 @@ var Migration = function () {
   }, {
     key: 'upgrade',
     value: function () {
-      var _ref2 = _asyncToGenerator(regeneratorRuntime.mark(function _callee2() {
+      var _ref2 = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee2() {
         return regeneratorRuntime.wrap(function _callee2$(_context2) {
           while (1) {
             switch (_context2.prev = _context2.next) {
@@ -110,7 +113,7 @@ var Migration = function () {
   }, {
     key: 'scrollSearch',
     value: function () {
-      var _ref3 = _asyncToGenerator(regeneratorRuntime.mark(function _callee3(index, type, query, options) {
+      var _ref3 = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee3(index, type, query, options) {
         var objects, opts, searchOptions, response;
         return regeneratorRuntime.wrap(function _callee3$(_context3) {
           while (1) {
@@ -197,7 +200,7 @@ var Migration = function () {
   }, {
     key: 'countHits',
     value: function () {
-      var _ref4 = _asyncToGenerator(regeneratorRuntime.mark(function _callee4(index, type, query) {
+      var _ref4 = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee4(index, type, query) {
         var searchOptions, response;
         return regeneratorRuntime.wrap(function _callee4$(_context4) {
           while (1) {

--- a/lib/migrations/migration_logger.js
+++ b/lib/migrations/migration_logger.js
@@ -8,9 +8,11 @@ var _createClass = function () { function defineProperties(target, props) { for 
 
 function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
 
+require("babel-polyfill");
 /**
  * A logger for migrations.
  */
+
 var MigrationLogger = function () {
 
   /**

--- a/lib/migrations/migration_runner.js
+++ b/lib/migrations/migration_runner.js
@@ -12,9 +12,12 @@ function _asyncToGenerator(fn) { return function () { var gen = fn.apply(this, a
 
 function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
 
+require("babel-polyfill");
+
 /**
  * A migration runner.
  */
+
 var MigrationRunner = function () {
   /**
    * Creates a new Migration runner.
@@ -111,7 +114,7 @@ var MigrationRunner = function () {
   }, {
     key: 'count',
     value: function () {
-      var _ref = _asyncToGenerator(regeneratorRuntime.mark(function _callee() {
+      var _ref = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee() {
         var loggingEnabled = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : true;
 
         var toUpgrade, warning, _iteratorNormalCompletion2, _didIteratorError2, _iteratorError2, _iterator2, _step2, migration, _count;
@@ -213,7 +216,7 @@ var MigrationRunner = function () {
   }, {
     key: 'upgrade',
     value: function () {
-      var _ref2 = _asyncToGenerator(regeneratorRuntime.mark(function _callee2() {
+      var _ref2 = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee2() {
         var migrations, maxIteration, iteration, upgradedTotal, upgradedThisIteration, totalCount, info, _iteratorNormalCompletion3, _didIteratorError3, _iteratorError3, _iterator3, _step3, migration, count;
 
         return regeneratorRuntime.wrap(function _callee2$(_context2) {

--- a/lib/test_utils/index_snapshot.js
+++ b/lib/test_utils/index_snapshot.js
@@ -14,7 +14,7 @@ function _asyncToGenerator(fn) { return function () { var gen = fn.apply(this, a
  * @return {Map} having the id's as keys and the hits as values.
  */
 exports.default = function () {
-  var _ref = _asyncToGenerator(regeneratorRuntime.mark(function _callee(client, index) {
+  var _ref = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee(client, index) {
     var response;
     return regeneratorRuntime.wrap(function _callee$(_context) {
       while (1) {

--- a/lib/test_utils/scenario_manager.js
+++ b/lib/test_utils/scenario_manager.js
@@ -51,7 +51,7 @@ var ScenarioManager = function () {
   _createClass(ScenarioManager, [{
     key: 'load',
     value: function () {
-      var _ref = _asyncToGenerator(regeneratorRuntime.mark(function _callee(scenario) {
+      var _ref = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee(scenario) {
         var self, _iteratorNormalCompletion, _didIteratorError, _iteratorError, _iterator, _step, bulk, _body, body, response;
 
         return regeneratorRuntime.wrap(function _callee$(_context) {
@@ -216,7 +216,7 @@ var ScenarioManager = function () {
   }, {
     key: 'unload',
     value: function () {
-      var _ref2 = _asyncToGenerator(regeneratorRuntime.mark(function _callee2(scenario) {
+      var _ref2 = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee2(scenario) {
         var indices, start, exists;
         return regeneratorRuntime.wrap(function _callee2$(_context2) {
           while (1) {
@@ -317,7 +317,7 @@ var ScenarioManager = function () {
      * @param {object} scenario The scenario to reload.
      */
     value: function () {
-      var _ref3 = _asyncToGenerator(regeneratorRuntime.mark(function _callee3(scenario) {
+      var _ref3 = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee3(scenario) {
         var self;
         return regeneratorRuntime.wrap(function _callee3$(_context3) {
           while (1) {
@@ -353,7 +353,7 @@ var ScenarioManager = function () {
      * Sends a delete all indices request
      */
     value: function () {
-      var _ref4 = _asyncToGenerator(regeneratorRuntime.mark(function _callee4() {
+      var _ref4 = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee4() {
         return regeneratorRuntime.wrap(function _callee4$(_context4) {
           while (1) {
             switch (_context4.prev = _context4.next) {

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "kibiutils",
-  "version": "1.8.0",
+  "version": "1.9.0",
   "description": "Small utility lib for the Kibi project",
   "main": "./lib/kibiutils.js",
   "scripts": {
@@ -29,6 +29,12 @@
     "url": "https://github.com/sirensolutions/kibiutils/issues"
   },
   "babel": {
+    "plugins": [
+      "add-module-exports",
+      "transform-async-generator-functions",
+      "transform-class-properties",
+      "transform-object-rest-spread"
+    ],
     "presets": [
       "es2015",
       "es2017"
@@ -36,21 +42,35 @@
   },
   "homepage": "https://github.com/sirensolutions/kibiutils",
   "devDependencies": {
-    "@elastic/eslint-config-kibana": "0.2.1",
-    "babel": "^6.23.0",
-    "babel-cli": "^6.24.0",
-    "babel-eslint": "^6.1.2",
-    "babel-polyfill": "^6.23.0",
-    "babel-preset-es2015": "^6.24.0",
-    "babel-preset-es2017": "^6.22.0",
-    "babel-register": "^6.24.0",
-    "babel-runtime": "5.8.38",
+    "@elastic/eslint-config-kibana": "0.6.1",
+    "@elastic/eslint-plugin-kibana-custom": "1.0.3",
+    "bluebird": "2.9.34",
+    "elasticsearch": "13.0.1",
+    "babel-cli": "6.18.0",
+    "babel-core": "6.21.0",
+    "babel-eslint": "7.2.3",
+    "babel-jest": "20.0.0",
+    "babel-loader": "6.2.10",
+    "babel-plugin-add-module-exports": "0.2.1",
+    "babel-plugin-transform-async-generator-functions": "6.24.1",
+    "babel-plugin-transform-class-properties": "6.24.1",
+    "babel-plugin-transform-object-rest-spread": "6.23.0",
+    "babel-polyfill": "6.20.0",
+    "babel-preset-es2015": "6.24.1",
+    "babel-preset-es2017": "6.24.1",
+    "babel-register": "6.18.0",
     "eslint": "3.11.1",
+    "eslint-plugin-babel": "4.1.1",
+    "eslint-plugin-import": "2.3.0",
+    "eslint-plugin-jest": "20.0.0",
     "eslint-plugin-mocha": "4.7.0",
+    "eslint-plugin-react": "7.0.1",
     "expect.js": "~0.3.1",
     "lodash": "3.10.1",
     "mocha": "2.5.3",
     "precommit-hook": "3.0.0",
-    "sinon": "1.17.2"
-  }
+    "sinon": "1.17.2",
+    "webpack": "github:elastic/webpack#fix/query-params-for-aliased-loaders"
+  },
+  "dependencies": {}
 }

--- a/src/__tests__/tests.js
+++ b/src/__tests__/tests.js
@@ -5,55 +5,55 @@ import _ from 'lodash';
 describe('Json traversing', function () {
   describe('getValuesAtPath', function () {
     it('non existing path', function () {
-      var values = kibiutils.getValuesAtPath({ aaa: 'bbb' }, [ 'not', 'in', 'there' ]);
+      const values = kibiutils.getValuesAtPath({ aaa: 'bbb' }, [ 'not', 'in', 'there' ]);
       expect(values).to.have.length(0);
     });
 
     it('non existing path in nested object', function () {
-      var json = {
+      const json = {
         aaa: {
           bbb: 123
         }
       };
-      var values = kibiutils.getValuesAtPath(json, [ 'aaa', 'ccc' ]);
+      const values = kibiutils.getValuesAtPath(json, [ 'aaa', 'ccc' ]);
       expect(values).to.have.length(0);
     });
 
     it('get value at level one', function () {
-      var values = kibiutils.getValuesAtPath({ aaa: 'bbb' }, [ 'aaa' ]);
+      const values = kibiutils.getValuesAtPath({ aaa: 'bbb' }, [ 'aaa' ]);
       expect(values).to.eql([ 'bbb' ]);
     });
 
     it('get nested value', function () {
-      var json = {
+      const json = {
         aaa: {
           bbb: 123
         }
       };
-      var values = kibiutils.getValuesAtPath(json, [ 'aaa', 'bbb' ]);
+      const values = kibiutils.getValuesAtPath(json, [ 'aaa', 'bbb' ]);
       expect(values).to.eql([ 123 ]);
     });
 
     it('value is an array', function () {
-      var json = {
+      const json = {
         aaa: [ 123, 456 ]
       };
-      var values = kibiutils.getValuesAtPath(json, [ 'aaa' ]);
+      const values = kibiutils.getValuesAtPath(json, [ 'aaa' ]);
       expect(values).to.eql([ 123, 456 ]);
     });
 
     it('nested value is an array', function () {
-      var json = {
+      const json = {
         aaa: {
           bbb: [ 123, 456 ]
         }
       };
-      var values = kibiutils.getValuesAtPath(json, [ 'aaa', 'bbb' ]);
+      const values = kibiutils.getValuesAtPath(json, [ 'aaa', 'bbb' ]);
       expect(values).to.eql([ 123, 456 ]);
     });
 
     it('multiple values are reachable via path', function () {
-      var json = {
+      const json = {
         aaa: [
           {
             bbb: 123
@@ -63,12 +63,12 @@ describe('Json traversing', function () {
           }
         ]
       };
-      var values = kibiutils.getValuesAtPath(json, [ 'aaa', 'bbb' ]);
+      const values = kibiutils.getValuesAtPath(json, [ 'aaa', 'bbb' ]);
       expect(values).to.eql([ 123, 456 ]);
     });
 
     it('multiple values with some that are arrays are reachable via path', function () {
-      var json = {
+      const json = {
         aaa: [
           {
             bbb: [ 123, 456 ]
@@ -78,12 +78,12 @@ describe('Json traversing', function () {
           }
         ]
       };
-      var values = kibiutils.getValuesAtPath(json, [ 'aaa', 'bbb' ]);
+      const values = kibiutils.getValuesAtPath(json, [ 'aaa', 'bbb' ]);
       expect(values).to.eql([ 123, 456, 789 ]);
     });
 
     it('nested array mix', function () {
-      var json = {
+      const json = {
         aaa: [
           {
             bbb: [
@@ -102,13 +102,13 @@ describe('Json traversing', function () {
           }
         ]
       };
-      var values = kibiutils.getValuesAtPath(json, [ 'aaa', 'bbb', 'ccc' ]);
+      const values = kibiutils.getValuesAtPath(json, [ 'aaa', 'bbb', 'ccc' ]);
       expect(values).to.eql([ 123, 456, 789 ]);
     });
 
     describe('nulls', function () {
       it('on level 1', function () {
-        var json = {
+        const json = {
           aaa: null
         };
         expect(kibiutils.getValuesAtPath(json, [ 'aaa' ])).to.have.length(0);
@@ -116,7 +116,7 @@ describe('Json traversing', function () {
       });
 
       it('on level 2', function () {
-        var json = {
+        const json = {
           aaa: {
             bbb: null
           }
@@ -126,7 +126,7 @@ describe('Json traversing', function () {
       });
 
       it('in array', function () {
-        var json = {
+        const json = {
           aaa: [
             123,
             null
@@ -136,7 +136,7 @@ describe('Json traversing', function () {
       });
 
       it('nested in array', function () {
-        var json = {
+        const json = {
           aaa: [
             {
               bbb: 123
@@ -152,14 +152,14 @@ describe('Json traversing', function () {
 
     describe('dotted field names', function () {
       it('on level 1', function () {
-        var json = {
+        const json = {
           'aaa.bbb': 123
         };
         expect(kibiutils.getValuesAtPath(json, [ 'aaa.bbb' ])).to.eql([ 123 ]);
       });
 
       it('nested', function () {
-        var json = {
+        const json = {
           'aaa.bbb': {
             ccc: 123
           }
@@ -171,7 +171,7 @@ describe('Json traversing', function () {
 
   describe('goToElement', function () {
     it('object in array', function () {
-      var json = {
+      const json = {
         aaa: [
           {
             bbb: 'ccc'
@@ -182,8 +182,8 @@ describe('Json traversing', function () {
         ]
       };
 
-      var ccc = 0;
-      var ddd = 0;
+      let ccc = 0;
+      let ddd = 0;
 
       kibiutils.goToElement(json, [ 'aaa', '0', 'bbb' ], function (nested) {
         if (nested === 'ccc') {
@@ -207,7 +207,7 @@ describe('Json traversing', function () {
     });
 
     it('object nested in array', function () {
-      var json = {
+      const json = {
         aaa: [
           {
             ccc: {
@@ -222,7 +222,7 @@ describe('Json traversing', function () {
     });
 
     it('nested object', function () {
-      var json = {
+      const json = {
         aaa: {
           bbb: [
             {
@@ -238,7 +238,7 @@ describe('Json traversing', function () {
     });
 
     it('nested arrays', function () {
-      var json = {
+      const json = {
         aaa: [
           'foo',
           [
@@ -252,8 +252,8 @@ describe('Json traversing', function () {
         ]
       };
 
-      var ccc = 0;
-      var ddd = 0;
+      let ccc = 0;
+      let ddd = 0;
 
       kibiutils.goToElement(json, [ 'aaa', '1', '0', 'bbb' ], function (nested) {
         if (nested === 'ccc') {
@@ -281,7 +281,7 @@ describe('Json traversing', function () {
 describe('Error handling', function () {
   describe('with goToElement', function () {
     it('bad path 1', function () {
-      var json = {
+      const json = {
         aaa: 'bbb'
       };
 
@@ -289,7 +289,7 @@ describe('Error handling', function () {
     });
 
     it('bad path 2', function () {
-      var json = {
+      const json = {
         aaa: 'bbb'
       };
 
@@ -300,13 +300,13 @@ describe('Error handling', function () {
 
 describe('uuid generation', function () {
   it('get uuid4', function () {
-    var uuid4 = kibiutils.getUuid4();
+    const uuid4 = kibiutils.getUuid4();
     expect(uuid4).to.match(/^[0-9a-f]{8}-[0-9a-f]{4}-4[0-9a-f]{3}-[0-9a-f]{4}-[0-9a-f]{12}$/);
   });
 });
 
 describe('slugifyId()', function () {
-  var fixtures = [
+  const fixtures = [
     ['test#test', 'test-hash-test'],
     ['test/test', 'test-slash-test'],
     ['test?test', 'test-questionmark-test'],
@@ -327,9 +327,9 @@ describe('slugifyId()', function () {
   ];
 
   _.each(fixtures, function (fixture) {
-    var msg = 'should convert ' + fixture[0] + ' to ' + fixture[1];
+    const msg = 'should convert ' + fixture[0] + ' to ' + fixture[1];
     it(msg, function () {
-      var results = kibiutils.slugifyId(fixture[0]);
+      const results = kibiutils.slugifyId(fixture[0]);
       expect(results).to.be(fixture[1]);
     });
   });
@@ -340,7 +340,7 @@ describe('slugifyId()', function () {
 });
 
 describe('doesQueryDependOnEntity()', function () {
-  var notDependentQueries = [
+  const notDependentQueries = [
     'SELECT * \n' +
     'FROM test \n' +
     '-- WHERE name = \'@doc[_source][github_id]@\'',
@@ -371,7 +371,7 @@ describe('doesQueryDependOnEntity()', function () {
     '}'
   ];
 
-  var dependentQueries = [
+  const dependentQueries = [
     'SELECT * \n' +
     'FROM test \n' +
     'WHERE name = \'@doc[_source][github_id]@\'',
@@ -394,10 +394,10 @@ describe('doesQueryDependOnEntity()', function () {
   ];
 
   it('should check queries for commented lines on activationQuery', function () {
-    var queries = [];
+    const queries = [];
 
-    for (var i = 0; i < notDependentQueries.length; i++) {
-      var query = {
+    for (let i = 0; i < notDependentQueries.length; i++) {
+      const query = {
         activationQuery: '',
         resultQuery: ''
       };
@@ -409,8 +409,8 @@ describe('doesQueryDependOnEntity()', function () {
   });
 
   it('should check queries for not commented lines on activationQuery', function () {
-    for (var i = 0; i < dependentQueries.length; i++) {
-      var query = {
+    for (let i = 0; i < dependentQueries.length; i++) {
+      const query = {
         activationQuery: '',
         resultQuery: ''
       };
@@ -421,10 +421,10 @@ describe('doesQueryDependOnEntity()', function () {
   });
 
   it('should check queries for commented lines on resultQuery', function () {
-    var queries = [];
+    const queries = [];
 
-    for (var i = 0; i < notDependentQueries.length; i++) {
-      var query = {
+    for (let i = 0; i < notDependentQueries.length; i++) {
+      const query = {
         activationQuery: '',
         resultQuery: ''
       };
@@ -436,8 +436,8 @@ describe('doesQueryDependOnEntity()', function () {
   });
 
   it('should check queries for not commented lines on resultQuery', function () {
-    for (var i = 0; i < dependentQueries.length; i++) {
-      var query = {
+    for (let i = 0; i < dependentQueries.length; i++) {
+      const query = {
         activationQuery: '',
         resultQuery: ''
       };

--- a/src/kibiutils.js
+++ b/src/kibiutils.js
@@ -1,4 +1,5 @@
 (function () {
+  require("babel-polyfill");
 
   function KibiUtils() {}
 
@@ -8,16 +9,16 @@
     // PRIVATE methods
     //
 
-    var _checkSingleQuery = function (query) {
+    const _checkSingleQuery = function (query) {
 
-      var entityRegex = /@doc\[.+?\]@/;
-      var multilineCommentRegex = /\/\*(.|[\r\n])*?\*\//g;
-      var singleLineRegex = /(-- |# |\/\/).*/g;
+      const entityRegex = /@doc\[.+?\]@/;
+      const multilineCommentRegex = /\/\*(.|[\r\n])*?\*\//g;
+      const singleLineRegex = /(-- |# |\/\/).*/g;
 
       // check for sparql and sql queries
-      var activationQueryCheck = query.activationQuery &&
+      const activationQueryCheck = query.activationQuery &&
         entityRegex.test(query.activationQuery.replace(multilineCommentRegex, '').replace(singleLineRegex, ''));
-      var resultQueryCheck = query.resultQuery &&
+      const resultQueryCheck = query.resultQuery &&
         entityRegex.test(query.resultQuery.replace(multilineCommentRegex, '').replace(singleLineRegex, ''));
       if (activationQueryCheck || resultQueryCheck) {
         // requires entityURI
@@ -34,7 +35,7 @@
           return true;
         }
 
-        var i;
+        let i;
         if (query.rest_params) {
           for (i = 0; i < query.rest_params.length; i++) {
             if (entityRegex.test(query.rest_params[i].value)) {
@@ -53,8 +54,8 @@
       return false;
     };
 
-    var doesQueryDependOnEntity = function (queries) {
-      for (var i = 0; i < queries.length; i++) {
+    const doesQueryDependOnEntity = function (queries) {
+      for (let i = 0; i < queries.length; i++) {
         if (_checkSingleQuery(queries[i])) {
           return true;
         }
@@ -70,14 +71,14 @@
      * @param path the path as an array
      * @returns an array with all the values reachable from path
      */
-    var getValuesAtPath = function (json, path) {
+    const getValuesAtPath = function (json, path) {
       if (!path || !path.length) {
         return [];
       }
 
-      var values = [];
+      const values = [];
 
-      var getValues = function (element, pathIndex) {
+      const getValues = function (element, pathIndex) {
         if (!element) {
           return;
         }
@@ -88,7 +89,7 @@
               throw new Error('The value at ' + path.join('.') + ' cannot be an object: ' + JSON.stringify(json, null, ' '));
             }
             if (element.constructor === Array) {
-              for (var i = 0; i < element.length; i++) {
+              for (let i = 0; i < element.length; i++) {
                 if (element[i]) {
                   if (element[i].constructor === Object) {
                     throw new Error('The value at ' + path.join('.') + ' cannot be an object: ' + JSON.stringify(json, null, ' '));
@@ -105,7 +106,7 @@
             getValues(element[path[pathIndex]], pathIndex + 1);
           }
         } else if (element.constructor === Array) {
-          for (var childi = 0; childi < element.length; childi++) {
+          for (let childi = 0; childi < element.length; childi++) {
             getValues(element[childi], pathIndex);
           }
         } else {
@@ -117,7 +118,7 @@
       return values;
     };
 
-    var _goToElement0 = function (json, path, ind, cb) {
+    const _goToElement0 = function (json, path, ind, cb) {
       // the path is created from splitting a string on PATH_SEPARATOR.
       // If that string is empty, then the element in the array
       // is an empty string.
@@ -131,7 +132,7 @@
           }
           _goToElement0(json[path[ind]], path, ind + 1, cb);
         } else if (json.constructor === Array) {
-          var arrayInd = parseInt(path[ind], 10);
+          const arrayInd = parseInt(path[ind], 10);
           _goToElement0(json[arrayInd], path, ind + 1, cb);
         } else {
           throw new Error('Unexpected JSON type: ' + JSON.stringify(json, null, ' '));
@@ -139,10 +140,10 @@
       }
     };
 
-    var slugifyId = function (id) {
+    const slugifyId = function (id) {
       if (id == null) return;
 
-      var trans = {
+      const trans = {
         '/' : '-slash-',
         '\\?' : '-questionmark-',
         '\\&' : '-ampersand-',
@@ -150,9 +151,9 @@
         '#' : '-hash-'
       };
 
-      for (var key in trans) {
+      for (const key in trans) {
         if (trans.hasOwnProperty(key)) {
-          var regex = new RegExp(key, 'g');
+          const regex = new RegExp(key, 'g');
           id = id.replace(regex, trans[key]);
         }
       }
@@ -165,7 +166,7 @@
     // Datasource type helpers
     //
 
-    var DatasourceTypes = {
+    const DatasourceTypes = {
       sqlite: 'sqlite',
       rest: 'rest',
       postgresql: 'postgresql',
@@ -176,7 +177,7 @@
       tinkerpop3: 'tinkerpop3'
     };
 
-    var isJDBC = function (type) {
+    const isJDBC = function (type) {
       switch (type) {
         case DatasourceTypes.sql_jdbc:
         case DatasourceTypes.sparql_jdbc:
@@ -186,7 +187,7 @@
       }
     };
 
-    var isSPARQL = function (type) {
+    const isSPARQL = function (type) {
       switch (type) {
         case DatasourceTypes.sparql_http:
         case DatasourceTypes.sparql_jdbc:
@@ -196,7 +197,7 @@
       }
     };
 
-    var isSQL = function (type) {
+    const isSQL = function (type) {
       switch (type) {
         case DatasourceTypes.mysql:
         case DatasourceTypes.sqlite:
@@ -225,8 +226,8 @@
 
       getUuid4: function () {
         return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function (c) {
-          var r = Math.random() * 16 | 0;
-          var v = c === 'x' ? r : (r & 0x3 | 0x8);
+          const r = Math.random() * 16 | 0;
+          const v = c === 'x' ? r : (r & 0x3 | 0x8);
           return v.toString(16);
         });
       },

--- a/src/migrations/migration.js
+++ b/src/migrations/migration.js
@@ -1,4 +1,5 @@
 import { has, defaults, merge } from 'lodash';
+require("babel-polyfill");
 
 /**
  * The base class for migrations.

--- a/src/migrations/migration_logger.js
+++ b/src/migrations/migration_logger.js
@@ -1,3 +1,4 @@
+require("babel-polyfill");
 /**
  * A logger for migrations.
  */

--- a/src/migrations/migration_runner.js
+++ b/src/migrations/migration_runner.js
@@ -1,4 +1,5 @@
 import { each, has } from 'lodash';
+require("babel-polyfill");
 
 /**
  * A migration runner.

